analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharp/CSharpExplodedGraph.cs

```diff
@@ -660,12 +660,12 @@ private ProgramState VisitVariableDesignation(VariableDesignationSyntaxWrapper v
                 var variableSymbol = SemanticModel.GetDeclaredSymbol(singleVariableDesignation);
                 var newSymbolicValue = SymbolicValue.Create();
                 newProgramState = SetNewSymbolicValueIfTracked(variableSymbol, newSymbolicValue, newProgramState);
+                // When the pattern is "x is Type t" we know that "t != null", hence (SV != null)
+                newProgramState = newProgramState.SetConstraint(newSymbolicValue, ObjectConstraint.NotNull);
 
                 if (singleVariable)
                 {
-                    // When the pattern is "x is Type t" we know that "t != null", hence (SV != null)
-                    newProgramState = newProgramState.SetConstraint(newSymbolicValue, ObjectConstraint.NotNull);
-                    newProgramState = newProgramState.PushValue(newSymbolicValue);
+                    newProgramState = newProgramState.PushValue(new SymbolicValue());
```

Comment:
This is the actual fix. We need to push the result of `value is Something` witch is a bool. At this stage, it has no SE-logical relation to the designated variable `value is Something newVariable` from SymbolicValue point of view. 

SV for `newVariable` is a side effect of this designation, but doesn't effect the actual result of `value is Something` condition. So it should not be pushed (same as the discard pattern above)